pkg/tools/otelcol-config/opamp.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/mikefarah/yq/v4/pkg/yqlib"
+)
+
+var errRemoteControlNotEnabled = errors.New("remote control not enabled")
+
+func SetOpAmpEndpointAction(ctx *actionContext) error {
+	endpoint := ctx.Flags.SetOpAmpEndpoint
+	if err := validateEndpoint(endpoint); err != nil {
+		return err
+	}
+	conf, err := ReadConfigDir(ctx.ConfigDir)
+	if err != nil {
+		return err
+	}
+
+	writer := ctx.WriteSumologicRemote
+	doc := conf.SumologicRemote
+
+	if len(doc) == 0 {
+		return fmt.Errorf("cannot set opamp endpoint: %s", errRemoteControlNotEnabled)
+	}
+
+	encoder := yqlib.YamlFormat.EncoderFactory()
+	decoder := yqlib.YamlFormat.DecoderFactory()
+	eval := yqlib.NewStringEvaluator()
+
+	expression := fmt.Sprintf(".extensions.opamp.endpoint = %q", endpoint)
+	result, err := eval.EvaluateAll(expression, string(doc), encoder, decoder)
+	if err != nil {
+		return fmt.Errorf("couldn't write installation token: %s", err)
+	}
```

Comment:
Are you sure this is the correct error message?

Reply:
Nope, this is incorrect, thanks!